### PR TITLE
Update for node iron lts

### PR DESCRIPTION
**Pull request by mattgoud**
### Upgrade node modules

- Upgrade some packages + audit fix (reduce some security warnings) to prepare for the transition to node 20
 (see [#37041](https://github.com/PrestaShop/PrestaShop/pull/37041) )
- fix style (due to bootstrap-touchspin upgrade )

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | upgrade node modules
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | related #34058
| Sponsor company   | Your company or customer's name goes here (if applicable).
| How to test?      | Indicate how to verify that this change works as expected.

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hello @mattgoud!

This is your first pull request on [classic-theme](https://github.com/PrestaShop/classic-theme/) repository of the PrestaShop project.

Thank you, and welcome to this Open Source community!

File: .hintrc
Comment (jolelievre) [inline .hintrc:1]: Shouldn't this be put in the `_dev` folder along with the `package.json` file and all the other configs?

File: _dev/css/components/products.scss
Comment (jolelievre) [inline _dev/css/components/products.scss]: Why remove this one? This seems like an UX change not required for the Node 20 compatibility no?
Unless this is related to the `bootstrap-touchspin` update you mentioned?

File: _dev/css/partials/_commons.scss
Comment (jolelievre) [inline _dev/css/partials/_commons.scss]: Same question as above, what is the link between this and the Node 20 compatibility?

File: _dev/js/cart.js
Comment (jolelievre) [inline _dev/js/cart.js]: Doesn't this change the style of all the icons?
Is it also because of `bootstrap-touchspin` update?

File: _dev/package.json
Comment (jolelievre) [inline _dev/package.json]: Ok so bootstrap version was not changed, which is critical here for backward compatibility of all the modules.

I wonder if `bootstrap-touchspin` should be updated. For the same reason, updating to a new major version maybe bring some breaking changes. Is it possible to keep the same major version? Or was it not possible because of Node 20 update?

**Review by jolelievre — DISMISSED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by florine2623** [comment]
No need functional testing :) ok to be merged

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Comment by jolelievre** [comment]
Thanks @mattgoud

**Pull request merged**
